### ISSUE TITLE
Mi Band 1A firmware downgrade fails

### ISSUE THREAD

**Issue by alex-mazzariol** [Freeyourgadget/Gadgetbridge#365]
I have a Mi Band 1A with firmware version 5.15.11.19, and I wanted to downgrade it to 5.15.7.14 because I'm having problems with the Bluetooth connection. I'm using GB 0.11.2.

When I select the Mili_1a.fw file (md5 is e81eb70e1f3582b27b82bca30f37d189) and load it in GB, I can confirm the version is correct. Once I start the upgrade (actually, downgrade) GB crashes on me.

In the log, after the file has been selected and the upgrade started, I find these lines:

```
20:39:27.764 [main] INFO  n.f.g.s.d.m.AbstractMi1FirmwareInfo - Only major version 1 is supported: 5
20:39:27.765 [main] INFO  n.f.g.s.d.m.Mi1FirmwareInfo - firmware not supported
20:39:27.773 [main] INFO  n.f.g.s.d.m.CompositeMiFirmwareInfo - unrecognized header
20:39:27.775 [main] INFO  n.f.g.s.d.m.Mi1SFirmwareInfo - firmware not supported
20:39:27.842 [main] DEBUG n.f.g.s.DeviceCommunicationService - Service startcommand: nodomain.freeyourgadget.gadgetbridge.devices.action.connect
20:39:27.915 [main] INFO  n.f.g.s.DeviceCommunicationService - Setting broadcast receivers to: true
20:39:29.770 [main] DEBUG n.f.g.s.DeviceCommunicationService - Service startcommand: nodomain.freeyourgadget.gadgetbridge.devices.action.install
20:39:29.771 [main] INFO  n.f.g.s.DeviceCommunicationService - will try to install app/fw
20:39:29.775 [main] INFO  n.f.g.i.GBDevice - Mark device as busy: fetch activity data
20:39:33.497 [main] INFO  n.f.g.Logging - Gadgetbridge version: 0.11.2
```

The last line is from after GB has been restarted after the crash.

As far as I can tell it looks like GB does not support any firmware version beyond 1; is it correct? If this is the case, how can I downgrade my 1A?

**Comment by d-p** [comment]
Hello,
some new findings on this topic.

the latest GB version 0.13.3 crash right before starting the flash for
Mi Band 1A (with white LEDs, no heartrate sensor) for the firmware
version 5.15.7.14 (recommended) at:
https://github.com/Freeyourgadget/Gadgetbridge/wiki/Mi-Band-Firmware-Information#515714-recommended-mifit-15912-1539

The GB version 0.9.8 works fine (I used this version because it was my
first GB version installed and I was almost sure that it was working).

Please note that downgrading the app for upgrading the firmware is a
multi step procedure if you want to preserve your data.

The user interface explain that the firware version is supported, the
log just before the crash says the opposite:

00:00:00.914 [GadgetBridge GATT Dispatcher] DEBUG n.f.g.s.b.TransactionBuilder - use NOTIFICATION
00:00:00.919 [main] INFO  n.f.g.s.d.m.AbstractMi1FirmwareInfo - Only major version 1 is supported: 5
00:00:00.920 [main] INFO  n.f.g.s.d.m.Mi1FirmwareInfo - firmware not supported
00:00:00.920 [main] INFO  n.f.g.s.d.m.CompositeMiFirmwareInfo - unrecognized header
00:00:00.920 [main] INFO  n.f.g.s.d.m.Mi1SFirmwareInfo - firmware not supported

Best,

**Comment by cpfeiffer** [comment]
The output you see is the probing of device and firmware and matching them. If you add a complete log, I can have a look at what's wrong.

I just successfully downgraded my 1A's firmware to that very same version without a problem.

We had a problem somewhere where some bands where detected as the wrong model. For that I'd need the log to confirm.

**Comment by cpfeiffer** [comment]
@alex-mazzariol Sorry for not answering earlier -- I completely missed this issue.

**Comment by sicherist** [comment]
Thank you for creating Gadgetbridge! I happly uninstalled that absurd Mi Fit-App ;)

I'm also facing the issue described here, trying to downgrade my 1As firmware. Log file attached.

[gadgetbridge log.txt](https://github.com/Freeyourgadget/Gadgetbridge/files/560386/gadgetbridge.log.txt)

**Comment by cpfeiffer** [comment]
Thanks for the log. That confirms that there is a new hw revision for Mi1A. We need to adjust our recognition code to that.

**Comment by sicherist** [comment]
With Gadgetbridge V 0.9.8 the downgrade worked however.

**Comment by cpfeiffer** [comment]
Yes, we probably had a less strict hw detection back then.

**Comment by cpfeiffer** [comment]
Please report back if this is fixed with 0.13.8.

**Comment by sicherist** [comment]
Yes the firmware downgrade worked with 0.13.8, thanks!
